### ISSUE TITLE
installation troubles

### ISSUE THREAD

**Issue by sblunt** [darthoctopus/reggae#10]
Hey folks! I'm one of the two JOSS reviewers for `reggae`. Sorry for the delay in getting started with this review! I hope I can help y'all out. I couldn't find your readthedocs page (but I see you're working on it), so for installation I followed your markdown file instructions.

I had trouble getting `reggae` installed in a fresh conda environment on my mac following the installation instructions, an issue which I've traced to the fact that the version of `pbjam` on pypi (1.0.0) is not up to date with the most recent github version (1.0.2), and `pbjam==1.0.0` fails to install. I was able to fix this by manually cloning and installing `pbjam==1.0.2` then retrying the `reggae` installation.

So, I see two possibilities:
1) encourage the `pbjam` folks to upload the most recent release to pypi
2) add this workaround of cloning and installing `pbjam` separately to your installation docs. If you go this route, I'd also suggest adding `pbjam>=1.0.2` to your requirements.txt. 

One more suggestion on installation: it's generally recommended to use `pip install .` in a git cloned repo rather than `python setup.py install`. For developers, you can recommend `pip install -e .` if desired-- this flag will make it so that changes to the local copy of the code are automatically propagated to the local installation.

https://github.com/openjournals/joss-reviews/issues/6588

**Comment by darthoctopus** [comment]
Hi Sarah,

We've restructured our requirements.txt to explicitly require pbjam 1.0.3, which is now on PyPI. We have also rewritten our installation instructions to include your suggestion to use pip rather than setup.py.

I have tested that the new instructions work on a fresh python3.10 environment --- unfortunately, pbjam currently has a dependency on a rather old version of scipy, and reggae won't be installable on newer versions of python until pbjam 2 is released. Please try again and let us know if you encounter any issues.

**Comment by sblunt** [comment]
Sounds great! I'd recommend adding a note that you require python<=3.10 to the install page as well. But otherwise the updated install page looks good to me.

**Comment by sblunt** [comment]
One more thing-- since jax is one of your dependencies, I think it's worth noting that some of your users may have trouble installing depending on their OS (see [here](https://jax.readthedocs.io/en/latest/installation.html#supported-platforms)), and you could potentially include a link to the jax install page under an "common installation problems FAQ" or soemthing like that

**Comment by darthoctopus** [comment]
Hi Sarah,

We've added both a note about python versions, and a section for FAQs with a reference to the jax OS support page, to the installation section of our documentation. Thanks for these suggestions.

**Comment by sblunt** [comment]
Ok, I was finally able to get the code working by downgrading my versions of `PyQt6-Qt6` and `PyQt6` to 6.4.0! I was getting a segfault using `PyQt6-Qt6`>=6.5 and `PyQt6`>=6.5. I'm not sure if I can make an actionable suggestion, other than it might be a good thing to add a suggestion along those lines to the installation FAQs?

**Comment by sblunt** [comment]
Oh, one more thing-- at one point in my debugging process, I tried running the code in Python 3.9 and ran into a different error. If you're also seeing that, then I'd suggest adding to the docs that you support Python 3.10 only.

**Comment by darthoctopus** [comment]
Hi Sarah,

Thanks for these notes.

- I'm able to reproduce your issue on Python 3.9 and have pushed a fix.
- I am unfortunately *not* able to reproduce your Qt issue — e.g. on both Python 3.10 and 3.9, I have been able to get our code to work on PyQt6 6.7.0. Could you please provide more information about your environment? This would be helpful for writing the entry in the FAQ that you suggest.

**Comment by sblunt** [comment]
Sure. Here's the full list of packages in the environment (python 3.10.0) with the error (output from `pip list`):

alabaster                     0.7.16
arviz                         0.12.1
astropy                       6.0.1
astropy-iers-data             0.2024.6.10.0.30.47
astroquery                    0.4.7
asttokens                     2.4.1
attrs                         23.2.0
autograd                      1.6.2
Babel                         2.15.0
backports.tarfile             1.2.0
beautifulsoup4                4.12.3
bleach                        6.1.0
bokeh                         3.4.1
cachetools                    5.3.3
certifi                       2024.6.2
cffi                          1.16.0
cftime                        1.6.4
charset-normalizer            3.3.2
contourpy                     1.2.1
corner                        2.2.2
cpnest                        0.11.5
cryptography                  42.0.8
cycler                        0.12.1
Cython                        3.0.10
decorator                     5.1.1
defusedxml                    0.7.1
deprecat                      2.1.1
dill                          0.3.8
docutils                      0.21.2
dynesty                       2.1.3
emcee                         3.1.4
exceptiongroup                1.2.1
executing                     2.0.1
fastjsonschema                2.19.1
fastprogress                  1.0.3
fbpca                         1.0
filelock                      3.14.0
fonttools                     4.53.0
future                        1.0.0
html5lib                      1.1
idna                          3.7
imagesize                     1.4.1
importlib_metadata            7.1.0
ipython                       8.25.0
jaraco.classes                3.4.0
jaraco.context                5.3.0
jaraco.functools              4.0.1
jax                           0.4.18
jaxlib                        0.4.18
jedi                          0.19.1
jeepney                       0.8.0
Jinja2                        3.1.4
joblib                        1.4.2
jsonschema                    4.22.0
jsonschema-specifications     2023.12.1
jupyter_client                8.6.2
jupyter_core                  5.7.2
jupyterlab_pygments           0.3.0
keyring                       25.2.1
kiwisolver                    1.4.5
lightkurve                    2.4.2
MarkupSafe                    2.1.5
matplotlib                    3.8.4
matplotlib-inline             0.1.7
memoization                   0.4.0
mistune                       3.0.2
ml-dtypes                     0.4.0
more-itertools                10.3.0
nbclient                      0.10.0
nbconvert                     7.16.4
nbformat                      5.10.4
nbsphinx                      0.9.3
netCDF4                       1.6.5
numpy                         1.22.1
oktopus                       0.1.2
opt-einsum                    3.3.0
packaging                     24.1
pandas                        2.0.3
pandocfilters                 1.5.1
parso                         0.8.4
patsy                         0.5.6
pbjam                         1.0.3
pexpect                       4.9.0
pillow                        10.3.0
pip                           24.0
platformdirs                  4.2.2
prompt_toolkit                3.0.47
ptyprocess                    0.7.0
pure-eval                     0.2.2
pycparser                     2.22
pyerfa                        2.0.1.4
Pygments                      2.18.0
pymc3                         3.11.5
pyparsing                     3.1.2
PyQt6                         6.7.0
PyQt6-Qt6                     6.7.2
PyQt6-sip                     13.6.0
python-dateutil               2.9.0.post0
pytz                          2024.1
pyvo                          1.5.2
PyYAML                        6.0.1
pyzmq                         26.0.3
referencing                   0.35.1
reggae                        0.0.1               /home/sblunt/reggae
requests                      2.32.3
rpds-py                       0.18.1
scikit-learn                  1.5.0
scipy                         1.7.3
SecretStorage                 3.3.3
semver                        3.0.2
setuptools                    69.5.1
six                           1.16.0
snowballstemmer               2.2.0
soupsieve                     2.5
Sphinx                        7.3.7
sphinxcontrib-applehelp       1.0.8
sphinxcontrib-devhelp         1.0.6
sphinxcontrib-htmlhelp        2.0.5
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.7
sphinxcontrib-serializinghtml 1.1.10
stack-data                    0.6.3
statsmodels                   0.14.1
Theano-PyMC                   1.1.2
threadpoolctl                 3.5.0
tinycss2                      1.3.0
tomli                         2.0.1
tornado                       6.4.1
tqdm                          4.66.4
traitlets                     5.14.3
typing_extensions             4.12.2
tzdata                        2024.1
uncertainties                 3.2.1
urllib3                       2.2.1
wcwidth                       0.2.13
webencodings                  0.5.1
wheel                         0.43.0
wrapt                         1.16.0
xarray                        2023.12.0
xarray-einstats               0.6.0
xyzservices                   2024.6.0
yabox                         1.1.0
zeta                          0a0
zipp                          3.19.2

Here's the OS info:

```NAME="Ubuntu"
VERSION="20.04.6 LTS (Focal Fossa)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 20.04.6 LTS"
VERSION_ID="20.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=focal
UBUNTU_CODENAME=focal
```

And here's the full stack trace for the error:

```/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: warning: libm.so.6, needed by /usr/lib/x86_64-linux-gnu/libblas.so, not found (try using -rpath or -rpath-link)
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: /usr/lib/x86_64-linux-gnu/libblas.so: undefined reference to `cabs@GLIBC_2.2.5'
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: /usr/lib/x86_64-linux-gnu/libblas.so: undefined reference to `cabsf@GLIBC_2.2.5'
collect2: error: ld returned 1 exit status
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: cannot find -lcblas: No such file or directory
collect2: error: ld returned 1 exit status
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: warning: libm.so.6, needed by /usr/lib/x86_64-linux-gnu/libblas.so, not found (try using -rpath or -rpath-link)
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: /usr/lib/x86_64-linux-gnu/libblas.so: undefined reference to `cabs@GLIBC_2.2.5'
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: /usr/lib/x86_64-linux-gnu/libblas.so: undefined reference to `cabsf@GLIBC_2.2.5'
collect2: error: ld returned 1 exit status
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: cannot find -lcblas: No such file or directory
collect2: error: ld returned 1 exit status
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: warning: libm.so.6, needed by /usr/lib/x86_64-linux-gnu/libblas.so, not found (try using -rpath or -rpath-link)
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: /usr/lib/x86_64-linux-gnu/libblas.so: undefined reference to `cabs@GLIBC_2.2.5'
/home/sblunt/miniconda3/envs/reggae/compiler_compat/ld: /usr/lib/x86_64-linux-gnu/libblas.so: undefined reference to `cabsf@GLIBC_2.2.5'
collect2: error: ld returned 1 exit status
qt.qpa.plugin: From 6.5.0, xcb-cursor0 or libxcb-cursor0 is needed to load the Qt xcb platform plugin.
qt.qpa.plugin: Could not load the Qt platform plugin "xcb" in "" even though it was found.
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.

Available platform plugins are: wayland-egl, minimalegl, offscreen, minimal, eglfs, vkkhrdisplay, wayland, linuxfb, vnc, xcb.

Aborted (core dumped)
```

**Comment by darthoctopus** [comment]
Hi Sarah,

Thanks for the description and the error dump. This seems like an issue we've run into previously on some linux systems where only Wayland is installed by default for user GUI sessions. This is unfortunately not something that we can reasonably anticipate, as it requires intervention through the package manager for some compiled dependencies to work (or, as you have found, using alternative versions for certain packages). I will add a note about this to the FAQs.

**Comment by sblunt** [comment]
Sounds good to me!

**Comment by darthoctopus** [comment]
Hi Sarah,

I've added an FAQ entry. Thanks for the heads up about your installation troubles!
